### PR TITLE
ci: Run integration test on GKE

### LINKED ISSUE #784: Run integration tests on GKE
Same as https://github.com/kinvolk/inspektor-gadget/issues/725 but for Google Kubernetes Engine (GKE).

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
This change allows running integration tests on GKE. It uses OIDC to authenticate with GCP similar to other cloud providers. Following are the successful runs:
- (amd64): https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/7901941522/job/21566707127
- (arm64): https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/7901941522/job/21566707426

In case you want to see logs for the tests that I have skipped (happy to have more feedback on this, also [potential fix](https://github.com/inspektor-gadget/inspektor-gadget/pull/2280#issuecomment-1939672769)):
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/7900722522/attempts/1#summary-21563030071
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/7887098920/attempts/2#summary-21523122919

Fixes #784

### CODE REVIEW

File: docs/devel/ci.md [hunk 1]
```diff
@@ -252,6 +252,25 @@ role (including the rull `arn:aws..` prefix). in a [repository
...
 
+## Run integration tests on GKE
+
+In order to run the integration tests on GKE on a fork, you need to set the following secrets:
+
+- `GKE_PROJECT`: The GCP project where the GKE cluster is located.
+- `GKE_SERVICE_ACCOUNT`: The service account used to authenticate to GCP.
+- `GKE_WORKLOAD_IDENTITY_PROVIDER`: The workload identity provider used to authenticate to GCP.
+
+these secret use OpenID Connect to authenticate GitHub Actions against GCP for more information on how it works
```
Comment (mauriciovasquezbernal) [inline docs/devel/ci.md:263]: secret -> secrets

File: integration/inspektor-gadget/top_ebpf_test.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func TestTopEbpf(t *testing.T) {
 	}
 
 	if *k8sDistro == K8sDistroGKECOS {
-		t.Skip("Skip running top ebpf gadget on GKE COS: see pull_request #2280")
+		t.Skip("Skip running top ebpf gadget on GKE COS: see pull request #2280")
```
Comment (mauriciovasquezbernal) [inline integration/inspektor-gadget/top_ebpf_test.go:73]: Please move these changes to the first commit in the PR. (Currently they're in the `docs: Add GKE support` one).

File: integration/inspektor-gadget/trace_network_test.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 
 func TestTraceNetwork(t *testing.T) {
 	if *k8sDistro == K8sDistroGKECOS {
-		t.Skip("Skip running top ebpf gadget on GKE COS: see pull_request #2280")
+		t.Skip("Skip running trace network on GKE COS: see pull request #2280")
```
Comment (mauriciovasquezbernal) [inline integration/inspektor-gadget/trace_network_test.go:29]: Same as above.

File: integration/inspektor-gadget/advise_networkpolicy_test.go [hunk 4]
```diff
@@ -25,6 +25,10 @@ import (
 )
 
 func TestAdviseNetworkpolicy(t *testing.T) {
+	if *k8sDistro == K8sDistroGKECOS {
+		t.Skip("Skip running advise network policy on GKE COS: see pull request #2280")
+	}
+
```
Comment (mauriciovasquezbernal) [inline integration/inspektor-gadget/advise_networkpolicy_test.go:31]: Same.

**Review by mauriciovasquezbernal — APPROVED** [review]
Very small minor comments comments. LGTM already. Thanks for working on this!

**Comment by mqasimsarfraz** [comment]
Thanks for the review!

**Pull request merged**